pep-0592.rst

```diff
@@ -0,0 +1,280 @@
+   checks:
+   ``Annotated[int, ValueRange(3, 10), ctype("char")] !=``\ Annotated[int,
+   ``ctype("char"),``\ ValueRange(3, 10)]```\`
+-  Nested ``Annotated`` types are flattened, with metadata ordered
+   starting with the innermost annotation:
+   ``Annotated[Annotated[int, ValueRange(3, 10)], ctype("char")] ==``\ Annotated[int,
+   ValueRange(3, 10), ctype(“char”)]`\`
+-  Duplicated annotations are not removed:
+   ``Annotated[int,``\ ValueRange(3, 10)\ ``] !=``\ Annotated[int,
+   ``ValueRange(3, 10)``, ``ValueRange(3, 10)``]```````\`
+-  ``Annotation`` can be used as a higher order aliases
+   (``Typevar T = ...; Vec = Annotated[List[Tuple[T, T]], MaxLen(10)]; V = Vec[int]``
+   then ``V ==``\ Annotated[List[Tuple[int, int]], MaxLen(10)]``)
+
+consuming annotations
+~~~~~~~~~~~~~~~~~~~~~
+
+Ultimately, the responsibility of how to interpret the annotations (if
+at all) is the responsibility of the tool or library encountering the
+``Annotated`` type. A tool or library encountering an ``Annotated`` type
+can scan through the annotations to determine if they are of interest
+(e.g., using ``isinstance``).
+
+**Unknown annotations:** When a tool or a library does not support
+annotations or encounters an unknown annotation it should just ignore it
+and treat annotated type as the underlying type. For example, if we were
+to add an annotation that is not an instance of ``struct.ctype`` to the
+annotation for name (e.g.,
+``Annotated[str, 'foo', struct.ctype("<10s")]``), the unpack method
+should ignore it.
+
+**Namespacing annotations:** We do not need namespaces for annotations
+since the class used by the annotations acts as a namespace.
+
+**Multiple annotations:** It’s up to the tool consuming the annotations
+to decide whether the client is allowed to have several annotations on
+one type and how to merge those annotations.
+
+Since the ``Annotated`` type allows you to put several annotations of
+the same (or different) type(s) on any node, the tools or libraries
+consuming those annotations are in charge of dealing with potential
+duplicates. For example, if you are doing value range analysis you might
+allow this::
+
+    T1 = Annotated[int, ValueRange(-10, 5)]
+    T2 = Annotated[T1, ValueRange(-20, 3)]
+
+Flattening nested annotations, this translates to::
+
+    ``T2 = Annotated[int, ValueRange(-10, 5), ValueRange(-20, 3)]``
+
+An application consuming this type might choose to reduce these
+annotations via an intersection of the ranges, in which case ``T2``
+would be treated equivalently to ``Annotated[int, ValueRange(-10, 3)]``.
+
+An alternative application might reduce these via a union, in which case
+``T2`` would be treated equivalently to
+``Annotated[int, ValueRange(-20, 5)]``.
+
+Consumers of specific annotations may decide to not support multiple
+annotations and throw an exception (e.g. in the example above the value
+range could throw on ``T2``).
+
+Interaction with ``get_type_hints``
```

Comment:
```suggestion
Interaction with ``get_type_hints()``
```
I like to include parentheses to emphasize functions. You can also update this elsewhere.